h/static/scripts/app-controller.coffee

```diff
@@ -38,9 +40,14 @@ module.exports = class AppController
     # Default sort
     $scope.sort = name: 'Location'
 
-    $scope.$on('groupFocused', (event) ->
-      $route.reload()
-    )
+    # Reload the view when the focused group changes or the
+    # list of groups that the user is a member of changes
+    groupChangeEvents = [events.SESSION_CHANGED, events.GROUP_FOCUSED];
+    groupChangeEvents.forEach((eventName) ->
+      $scope.$on(eventName, (event) ->
+        $route.reload()
```

Comment:
So the `SESSION_CHANGED` event will fire on login/logout, which means I think we are probably calling `$route.reload()` twice on login/logout now. Or, if we're not, it's only because of a race... You might be able to remove [some if not all of this `$watch`?](https://github.com/hypothesis/h/pull/2591/files#diff-e394c98a9a7311a619855172ca940fa6R79).